docs/typescript/reference/dbos-class.md

```diff
@@ -0,0 +1,957 @@
+DBOS.knexClient: Knex
+DBOS.typeORMClient: TypeORMEntityManager
+DBOS.drizzleClient: DrizzleClient
+```
+
+For more details, see:
+* [Drizzle](../tutorials/using-drizzle.md)
+* [Knex](../tutorials/using-knex.md)
+* [Prisma](../tutorials/using-prisma.md)
+* [TypeORM](../tutorials/using-typeorm.md)
+
+### `@DBOS.step`
+`@DBOS.step` registers a function as a DBOS step.  Such functions are a key building block of DBOS's [reliable workflows](../tutorials/workflow-tutorial.md).
+The result of each invocation of a function decorated with `@DBOS.step` is stored in the DBOS system database.  This checkpoint of the execution state allows function calls to be skipped during workflow replay, if the step is known to have completed previously.
+
+```typescript
+@DBOS.step(config?: StepConfig)
+```
+
+Functions decorated with `@DBOS.step` may be called directly, either inside or outside of workflows.  Functions decorated with `@DBOS.step` must be class members, and may be either `static` or [instance methods](#decorating-instance-methods).
+
+`@Step()` takes an optional `StepConfig`, which allows a number of step properties to be specified:
+
+```typescript
+export interface StepConfig {
+  retriesAllowed?: boolean; // Should failures be retried? (default true)
+  intervalSeconds?: number; // Seconds to wait before the first retry attempt (default 1).
+  maxAttempts?: number;     // Maximum number of retry attempts (default 3). If errors occur more times than this, throw an exception.
+  backoffRate?: number;     // Multiplier by which the retry interval increases after a retry attempt (default 2).
+}
+```
+
+Example:
+```typescript
+@DBOS.step()
+static async sendToServer(valueToSend: string) {
+  ...
+}
+```
+
+## Accessing Application Functions Requiring Context Arguments
+Prior versions of the DBOS SDK were based on functions that took a [`context`](./contexts.md#dboscontext) as the first argument.  It is possible to call these old-style step and transaction functions from new workflows via the `DBOS.invoke` syntax:
+
+```typescript
+DBOS.invoke<T extends ConfiguredInstance>(targetInst: T): InvokeFuncsInst<T>;
+DBOS.invoke<T extends object>(targetClass: T): InvokeFuncs<T>
+```
+
+Example:
+```typescript
+class MyClass {
+  // Older dercorator, function takes `ctx` for SQL access
+  @Transaction()
+  static async transactionFunction(ctx: TransactionContext, arg: string) {
+    //...
+  }
+}
+
+// Call with `DBOS.invoke`
+const res = DBOS.invoke(MyClass).transactionFunction('arg');
+```
+
+## Sleeping
+
+Within a DBOS workflow, waiting or sleeping should not be done using standard system functions, as these will not be skipped on workflow replay.  Instead, the "durable sleep" functions below should be used.  The wakeup time will be stored in the database when the function is first called, and if the workflow is re-executed, it will not oversleep.
+
+```typescript
+DBOS.sleepms(durationMS: number): Promise<void>
+DBOS.sleep(durationSec: number): Promise<void>
```

Comment:
Is now a good time to change `DBOS.sleep` to use millisecond by default? It's a new interface anyways...

Reply:
PR for you: https://github.com/dbos-inc/dbos-transact-ts/pull/674

Reply:
Approved!